zerver/lib/actions.py

```diff
@@ -718,6 +719,22 @@ def do_create_user(
     # Note that for bots, the caller will send an additional event
     # with bot-specific info like services.
     notify_created_user(user_profile)
+
+    system_user_group = get_system_user_group_for_user(user_profile)
+    UserGroupMembership.objects.create(user_profile=user_profile, user_group=system_user_group)
+    do_send_user_group_members_update_event("add_members", system_user_group, [user_profile.id])
+
+    if user_profile.role == UserProfile.ROLE_MEMBER and not user_profile.is_provisional_member:
+        full_members_system_group = UserGroup.objects.get(
+            name="@role:fullmembers", realm=user_profile.realm, is_system_group=True
+        )
+        UserGroupMembership.objects.create(
+            user_profile=user_profile, user_group=full_members_system_group
+        )
+        do_send_user_group_members_update_event(
+            "add_members", full_members_system_group, [user_profile.id]
+        )
```

Comment:
I guess ideally this would also be in a `transaction.atomic` with the user creation.  Doing so perfectly may require reworking this function to use `transaction.on_commit` for all the events it sends, etc., but we can potentially be a bit sloppier.  See `do_delete_message` for an example of this.

It'd be pretty annoying if these got out of sync, so I think it's worth at least ensuring that the `UserGroupMembership.objects.create` queries are in the main transaction.